warehouse/templates/manage/publishing.html

```diff
@@ -0,0 +1,129 @@
+{#
+ # Licensed under the Apache License, Version 2.0 (the "License");
+ # you may not use this file except in compliance with the License.
+ # You may obtain a copy of the License at
+ #
+ # http://www.apache.org/licenses/LICENSE-2.0
+ #
+ # Unless required by applicable law or agreed to in writing, software
+ # distributed under the License is distributed on an "AS IS" BASIS,
+ # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ # See the License for the specific language governing permissions and
+ # limitations under the License.
+-#}
+
+{% extends "manage_project_base.html" %}
+
+{% set active_tab = 'publishing' %}
+
+{% block title %}
+  {% trans %}OIDC provider management{% endtrans %}
+{% endblock %}
+
+{% block main %}
+{% if testPyPI %}
+{% set title = "TestPyPI" %}
+{% else %}
+{% set title = "PyPI" %}
+{% endif %}
+
+{% macro provider_row(provider) -%}
+<form id="delete-provider-{{ provider.id }}" method="POST" action="{{ request.route_path('manage.project.settings.oidc', project_name=project.name) }}">
+  <input name="csrf_token" type="hidden" value="{{ request.session.get_csrf_token() }}">
+  <input name="provider_id" type="hidden" value="{{ provider.id }}">
+</form>
+
+<tr>
+  <td scope="row">
+    {{ provider.provider_name }}
+  </td>
+  <td scope="row">
+    {{ provider|string }}
+  </td>
+  <td scope="row">
+    <input form="delete-provider-{{ provider.id }}" type="submit" value="{% trans %}Remove{% endtrans %}" class="button button--danger"></input>
+  </td>
+</tr>
+{%- endmacro %}
+
+<div class="horizontal-section">
+  <div class="site-container">
+    <h1 class="page-title">{% trans %}OIDC provider management{% endtrans %}</h1>
+
+    <p>
+      {% trans trimmed %}
+        OIDC provides a flexible, credential-free mechanism for delegating
+        publishing authority for a PyPI package to a third party service,
+        like GitHub Actions.
+      {% endtrans %}
+    </p>
+
+    <p class="no-bottom-padding">
+      {% trans trimmed %}
+        PyPI projects can use trusted OIDC providers to automate their release
+        processes, without having to explicitly provision or manage API tokens.
+      {% endtrans %}
+    </p>
+
+    <h2 class="no-bottom-padding">{% trans %}Add a new provider{% endtrans %}</h2>
+
+    <h3>GitHub</h3>
+
```

Comment:
We should link to https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/about-security-hardening-with-openid-connect somewhere, probably here.

Reply:
5ed70bc4